### ISSUE TITLE
00steam.rpy 'NoneType' object has no attribute 'encode' when using achievement.sync() and Steam Client Open

### ISSUE THREAD

**Issue by danac78** [renpy/renpy#5457]
```

I'm sorry, but an uncaught exception occurred.

While running game code:
  File "game/scenes/d1/eric_day1.rpy", line 1609, in script
    $SteamAchieveCheck("ACH_GOINGCRUISING")
  File "game/scenes/d1/eric_day1.rpy", line 1609, in <module>
    $SteamAchieveCheck("ACH_GOINGCRUISING")
  File "game/defines/steam.rpy", line 93, in SteamAchieveCheck
    achievement.sync()
  File "renpy/common/00achievement.rpy", line 279, in sync
    if not i.has(a):
  File "renpy/common/00steam.rpy", line 793, in has
    return steam.get_achievement(name)
  File "renpy/common/00steam.rpy", line 91, in get_achievement
    if not steamapi.SteamUserStats().GetAchievement(name.encode("utf-8"), byref(rv)):
AttributeError: 'NoneType' object has no attribute 'encode'

-- Full Traceback ------------------------------------------------------------

Full traceback:
  File "game/scenes/d1/eric_day1.rpy", line 1609, in script
    $SteamAchieveCheck("ACH_GOINGCRUISING")
  File "F:\testrenpy\renpy-8.2.0-sdk\renpy\ast.py", line 823, in execute
    renpy.python.py_exec_bytecode(self.code.bytecode, self.hide, store=self.store)
  File "F:\testrenpy\renpy-8.2.0-sdk\renpy\python.py", line 1178, in py_exec_bytecode
    exec(bytecode, globals, locals)
  File "game/scenes/d1/eric_day1.rpy", line 1609, in <module>
    $SteamAchieveCheck("ACH_GOINGCRUISING")
  File "game/defines/steam.rpy", line 93, in SteamAchieveCheck
    achievement.sync()
  File "renpy/common/00achievement.rpy", line 279, in sync
    if not i.has(a):
  File "renpy/common/00steam.rpy", line 793, in has
    return steam.get_achievement(name)
  File "renpy/common/00steam.rpy", line 91, in get_achievement
    if not steamapi.SteamUserStats().GetAchievement(name.encode("utf-8"), byref(rv)):
AttributeError: 'NoneType' object has no attribute 'encode'

Windows-10-10.0.22631 AMD64
Ren'Py 8.2.0.24012702
Fire Of Life: New Day  0.46s.20240406
Sat Apr  6 06:44:05 2024

```
I have yet to go further back with the version. The code for SteamAchieveCheck is:

```
def SteamAchieveCheck(nameofAchieve):
        if not achievement.has(nameofAchieve):
            achievement.grant(nameofAchieve)
        if "s" in config.version:            
            achievement.sync()
```
I call the function with a string name of the achievement (either with $ or Function() in screens), it checks if it is not already there and grant it. If s (for steam) it runs the sync. I do not know when this problem started as I had only recently had to redownload SteamAPI through the launcher.

Additional Information:
I commented out achievement.sync() and it is not longer an issue.

**Comment by danac78** [comment]
I have tried restarting and reinstall Steam to see if it was a PEBKAC issue..but it still persists. :)

**Comment by renpytom** [comment]
It looks like the issue was that None somehow got into the list of granted achievements. I've pushed some additional checks for this to the nightly fix branch, so tomorrow's builds should fix your problem. (It should clean invalid values out of the list of granted achievements.) 

You'd get this error if you ever had: 

```
$ achivevement.grant(None)
```

even indirectly, like through a function that returns None or a variable that was bound to None.

**Comment by danac78** [comment]
@renpytom So, it was likely that at some point, $SteamAchieveCheck("") happened..or Function(SteamAchieveCheck,"") happened when I was testing new content on my end, but because my Steam would see that, I was the only one affected because the rest have a proper string passing as an argument.

**Comment by danac78** [comment]
@renpytom The Fix (unless I downloaded it wrong at 3:26 Arizona time, it is possible) didn't seem to work. However, I did achievement_clear_all() (because it doesn't matter if I have the achievements since they will be granted again when I test), which is now resolved. Thanks for pointing me in the right direction. Is there a function that would list the achievements for diagnostic purposes in case None happens again?
